go/cmd/dolt/commands/sql.go

```diff
@@ -684,6 +694,38 @@ func sqlDelete(ctx context.Context, dEnv *env.DoltEnv, root *doltdb.RootValue, u
 	return result.Root, nil
 }
 
+// Checks if the query is a naked delete and then deletes all rows if so
+func sqlCheckThenDeleteAllRows(ctx context.Context, root *doltdb.RootValue, s *sqlparser.Delete) (*doltdb.RootValue, bool) {
+	if s.Where == nil && s.Limit == nil && s.Partitions == nil && len(s.TableExprs) == 1 {
+		if ate, ok := s.TableExprs[0].(*sqlparser.AliasedTableExpr); ok {
+			if ste, ok := ate.Expr.(sqlparser.TableName); ok {
+				tName := ste.Name.String()
+				table, ok, err := root.GetTable(ctx, tName)
+				if err == nil && ok {
+					emptyMap, err := types.NewMap(ctx, root.VRW())
+					if err != nil {
+						return nil, false
+					}
+					newTable, err := table.UpdateRows(ctx, emptyMap)
+					if err != nil {
+						return nil, false
+					}
+					newRoot, err := doltdb.PutTable(ctx, root, root.VRW(), tName, newTable)
+					if err != nil {
+						return nil, false
+					}
+					// This is so the results are displayed as expected
+					if _, sqlSch, rowIter, err := sqlNewEngine(fmt.Sprintf("select COUNT(*) as updated from `%v`", tName), root); err == nil {
```

Comment:
This is going to be expensive -- it loads the entire table of data just to count it. You should get the number of rows by calling .size() or similar on the table map